ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -955,6 +953,7 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
               <Dropdown style={{width: '14rem'}}
                         value={this.state.workspace.billingAccountName}
                         options={this.buildBillingAccountOptions()}
+                        disabled={freeTierCreditsBalance < 0.0 && !enableBillingUpgrade}
```

Comment:
The user can do nothing if Billing Upgrade is disabled and they have no free credits.